test/lib/browsers/base.ts

```diff
@@ -54,36 +45,43 @@ export abstract class BrowserInterface implements PromiseLike<any> {
   ): Promise<void>
   abstract close(): Promise<void>
 
-  abstract elementsByCss(selector: string): BrowserInterface[]
-  abstract elementByCss(selector: string): BrowserInterface
-  abstract elementById(selector: string): BrowserInterface
-  abstract touchStart(): BrowserInterface
-  abstract click(opts?: { modifierKey?: boolean }): BrowserInterface
-  abstract keydown(key: string): BrowserInterface
-  abstract keyup(key: string): BrowserInterface
-  abstract type(text: string): BrowserInterface
-  abstract moveTo(): BrowserInterface
-  // TODO(NEXT-290): type this correctly as awaitable
+  abstract elementsByCss(
+    selector: string
+  ): BrowserInterface<any[]> & Promise<any[]>
+  abstract elementByCss(selector: string): BrowserInterface<any> & Promise<any>
```

Comment:
Can we make the awaited version meaningful?
```suggestion
  abstract elementByCss(selector: string): BrowserInterface<any> & Promise<TheTypeForElementHandles>
```

Reply:
This can be said for any of those methods. As I wrote in the PR description:
> I'm not happy about using so many `any`s here, but this is due to having this abstract BrowserInterface class that is not supposed to know about the actual testing framework's types. [...] As a next step, either we define proper common types for those, or we merge the BrowserInterface and Playwright classes and just expose the Playwright types, since Selenium is not used anymore.

So are you suggesting we define a playwright-agnostic element handle type, or expose the playwright type?

Reply:
A union type would imply to me that consumers would potentially need to act differently based on what kind of handle they get. I don't think this is the intend with the abstract class, which I think tries to hide these testing framework differences from you. So consumers should probably be limited to using the intersection of all handle types. This would mean we could just as well define a common/shared/agnostic type, no?

Reply:
> So consumers should probably be limited to using the intersection of all handle types.

That's effectively what you'd get. But it would allow helpers to fork their implementation based on the testing framework that's currently enabled.